File: frontend/src/api/importPreset.ts
/**
 * Import Preset API client
 */
import request from './request'

export interface Preset {
  id: number
  category: string
  value: string
  is_default: boolean
  sort_order: number
  remark: string | null
}

export interface PresetCreate {
  category: string
  value: string
  is_default?: boolean
  sort_order?: number
  remark?: string
}

export interface PresetUpdate {
  value?: string
  sort_order?: number
  remark?: string
}

export interface SyncResult {
  location: number
  owner: number
  business_system: number
}

export const presetApi = {
  list: (category?: string, q?: string) =>
    request.get<any, Preset[]>('/api/import-presets', { params: { category, q } }),

  categories: () =>
    request.get<any, { key: string; label_key: string }[]>('/api/import-presets/categories'),

  create: (data: PresetCreate) =>
    request.post<any, Preset>('/api/import-presets', data),

  update: (id: number, data: PresetUpdate) =>
    request.put<any, Preset>(`/api/import-presets/${id}`, data),

  setDefault: (id: number) =>
    request.patch<any, Preset>(`/api/import-presets/${id}/default`),

  remove: (id: number) =>
    request.delete(`/api/import-presets/${id}`),

  syncFromAssets: () =>
    request.post<any, SyncResult>('/api/import-presets/sync-from-assets'),
}

File: backend/app/schemas/import_preset.py
"""
Import Preset Pydantic schemas
"""
from enum import Enum

from pydantic import BaseModel, Field


class PresetCategory(str, Enum):
    location = "location"
    owner = "owner"
    business_system = "business_system"


class PresetCreate(BaseModel):
    category: PresetCategory
    value: str = Field(min_length=1, max_length=255)
    is_default: bool = False
    sort_order: int = 0
    remark: str | None = None


class PresetUpdate(BaseModel):
    value: str | None = Field(default=None, min_length=1, max_length=255)
    sort_order: int | None = None
    remark: str | None = None


class PresetOut(BaseModel):
    id: int
    category: PresetCategory
    value: str
    is_default: bool
    sort_order: int
    remark: str | None

    model_config = {"from_attributes": True}


class SyncResult(BaseModel):
    location: int
    owner: int
    business_system: int
